modules/social_features/social_event/modules/social_event_addtocal/src/Plugin/SocialAddToCalendarBase.php

```diff
@@ -69,7 +69,9 @@ public function getIcon(): string {
       ->getPath('social_event_addtocal');
 
     $default_icon = $module_path . '/assets/icons/default-calendar.svg';
-    $plugin_icon = $module_path . '/assets/icons/' . $this->pluginDefinition['id'] . '.svg';
+    $plugin_icon = empty($this->pluginDefinition['id'])
+      ? $default_icon
+      : $module_path . '/assets/icons/' . $this->pluginDefinition['id'] . '.svg';
 
     return '/' . (file_exists($plugin_icon) ? $plugin_icon : $default_icon);
```

Comment:
I think we need to rewrite this also now, because in your case plugin_icon is never empty anymore.

Reply:
In the function return we are checking if file-exist and it can happen yet, but I can put together with previous if, what do you think?

Reply:
yes, in the old situation if the file exist we return $plugin_icon, and otherwise $default_icon.

But in the new one, it will never return $default_icon right? as we put that value in the $plugin_icon, so it will always return this $plugin_icon as it will always exist? Or is that a situation i'm not seeing where the return will return $default_icon still?

Reply:
If the image continue existing at database, but locally the file is missing, then we will use the default-icon.

Reply:
Not sure if I understand correctly, because you are now putting the default_icon in the plugin_icon it will always exist right? I don't see how the local file can not exist in our scenario. 

`    $plugin_icon = empty($this->pluginDefinition['id'])
      ? $module_path . '/assets/icons/default-calendar.svg'
      : $module_path . '/assets/icons/' . $this->pluginDefinition['id'] . '.svg';

    return '/' . $plugin_icon;`

so like